toolkit/scripts/sodiff/mariner-sodiff.sh

```diff
@@ -9,41 +9,93 @@ repo_file_path="$2"
 mariner_version="$3"
 sodiff_out_dir="$4"
 sodiff_log_file="${sodiff_out_dir}/sodiff.log"
+current_os=$(cat /etc/os-release | grep ^ID | cut -d'=' -f2)
+temporary_repo_file="file.repo"
 
 # Setup output dir
 mkdir -p "$sodiff_out_dir"
+# Empty the log file
+echo > "$sodiff_log_file"
 
-# Prepare mariner/ubuntu compatibility calls
+function makecache_with_common {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+        # Cache RPM metadata
+        >/dev/null dnf $common_options -y makecache
+    else
+        # Ubuntu uses repoquery command from yum-utils
+        DNF_COMMAND=
```

Comment:
I'd move setting `DNF_COMMAND` to a separate piece of code run only once. The variable is not used in this scope but outside of it. It makes the code a bit harder to follow.